gen/service.go

```diff
@@ -117,7 +112,26 @@ func functionHelper(g Generator, f *compile.FunctionSpec) error {
 						<unreserve .Name> <typeReferencePtr .Type>,
 					<end>
 				<end>
-			) *<$name>Args
+        `,
+		f,
+		TemplateFunc("unreserve", func(n string) string {
```

Comment:
maybe rename `unreserve` to `goArgName` or something, the template just wants a valid go name